src/Akka.Streams.Kafka/Helpers/Control.cs

```diff
@@ -41,95 +39,143 @@ public interface IControl
         Task<TResult> DrainAndShutdown<TResult>(Task<TResult> streamCompletion);
     }
 
+    /// <summary>
+    /// Helper class for creating a <see cref="DrainingControl{T}"/> instances.
+    /// </summary>
+    public static class DrainingControl
+    {
+        public static DrainingControl<T> Create<T>(IControl control, Task<T> streamCompletion) =>
+            DrainingControl<T>.Create(control, streamCompletion);
+
+        /// <summary>
+        /// Stop producing messages from the `Source`, wait for stream completion
+        /// and shut down the consumer `Source` so that all consumed messages
+        /// reach the end of the stream.
+        /// Failures in stream completion will be propagated, the source will be shut down anyway.
+        /// </summary>
+        internal static async Task<TResult> DrainAndShutdownDefaultAsync<TResult>(this IControl control,
+            Task<TResult> streamCompletion)
+        {
+            TResult result;
+
+            try
```

Comment:
Had to re-structure the `Task` and `await` composition significantly here in order to get `ControlSpec` to pass.